src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -0,0 +1,851 @@
+    }}
+}}";
+
+            await TestCS(csInput);
+        }
+
+        [Fact]
+        public async Task TestAbstractClass()
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+
+    class [|Program|] : AbClass
+    {
+        static void Main(string[] args)
+        {
+            Program prog = new Program();
+            prog.Bar();
+            [|prog.FooBar()|];
+            [|prog.BarImplemented()|];
+        }
+
+        public override void [|Bar|]()
+        {
+            throw new NotImplementedException();
+        }
+
+        [RequiresPreviewFeatures]
+        public override void FooBar()
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    [RequiresPreviewFeatures]
+    public abstract class AbClass
+    {
+        [RequiresPreviewFeatures]
+        public abstract void Bar();
+
+        [RequiresPreviewFeatures]
+        public abstract void FooBar();
+
+        [RequiresPreviewFeatures]
+        public void BarImplemented() => throw new NotImplementedException();
+    }
+}";
+
+            await TestCS(csInput);
+        }
+
+        [Fact]
+        public async Task TestUnmarkedPreviewProperty()
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+
+    class Program : IProgram
+    {
+        static void Main(string[] args)
+        {
+            new Program();
+        }
+
+        public bool [|MarkedPropertyInInterface|] { get => throw new NotImplementedException(); set => throw new NotImplementedException(); } // [] if not opted in yet
+    }
+
+    public interface IProgram
+    {
+        [RequiresPreviewFeatures]
+        bool MarkedPropertyInInterface { get; set; }
+    }
+}
+
+    ";
+
+            await TestCS(csInput);
+        }
+
+        [Fact]
+        public async Task TestUnmarkedPreviewInterface()
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+
+    class Program : IProgram
+    {
+        static void Main(string[] args)
+        {
+            new Program();
+        }
+
+        public void [|MarkedMethodInInterface|]()
+        {
+            throw new NotImplementedException();
+        }        
+    }
+
+    public interface IProgram
+    {
+        [RequiresPreviewFeatures]
+        void MarkedMethodInInterface();
+    }
+}
+
+    ";
+
+            await TestCS(csInput);
+
+        }
+
+        // TODO: The following test cannot be activated yet because it requires preview roslyn bits
+        //        [Fact]
+        //        public async Task TestPreviewLanguageFeatures()
+        //        {
+        //            var csInput = @" 
+        //using System.Runtime.Versioning; using System;
+        //namespace Preview_Feature_Scratch
+        //{"+
+        //@"
+
+        //    class Program : IProgram
```

Comment:
I'd uncomment it and use `[Fact(Skip = "Reason")]`. That way it's easier to track enabling the test.